plugins/database/mongodb/mongodb.go

```diff
@@ -248,19 +247,24 @@ func (m *MongoDB) RevokeUser(ctx context.Context, statements dbplugin.Statements
 		db = "admin"
 	}
 
-	err = session.DB(db).RemoveUser(username)
+	dropUserCmd := &dropUserCommand{
+		Username:     username,
+		WriteConcern: writeconcern.New(writeconcern.WMajority()),
+	}
+
+	result := client.Database(db).RunCommand(ctx, dropUserCmd, nil)
 	switch {
-	case err == nil, err == mgo.ErrNotFound:
-	case err == io.EOF, strings.Contains(err.Error(), "EOF"):
+	case result.Err() == nil:
+	case result.Err() == io.EOF, strings.Contains(result.Err().Error(), "EOF"):
 		if err := m.Close(); err != nil {
 			return errwrap.Wrapf("error closing EOF'd mongo connection: {{err}}", err)
 		}
-		session, err := m.getConnection(ctx)
+		client, err := m.getConnection(ctx)
 		if err != nil {
 			return err
 		}
-		err = session.DB(db).RemoveUser(username)
-		if err != nil {
+		result = client.Database(db).RunCommand(ctx, dropUserCmd, nil)
+		if result.Err() != nil {
 			return err
```

Comment:
These operations (Create, Revoke, ...) have a lot of repeated structure. Maybe there's a reasonable refactoring to extract a reusable core? If not, or if too complicated, then it's ok as is.

Reply:
I think this was referred to the `switch` statement which I've removed now, right?